### ISSUE TITLE
Support for sklearn/XGBoost methods

### ISSUE THREAD

**Issue by adubowski** [Ighina/CERTIFAI#2]
Do you plan to support also non-DL ML frameworks like sklearn and XGBoost?

**Comment by Ighina** [comment]
Hello, the main class should already support sklearn style models and therefore also the gradient boosting machines included in sklearn. Just be sure to select pytorch=False and classification=False when calling the various methods of the class (e.g. fit, etc.)

**Comment by adubowski** [comment]
Thank you, it works after some adjustments (e.g., 'torch' -> 'pytorch'), although I have to say the classification flag is not very intuitive since my task is a classification task with a sklearn DecisionTreeClassifier. 

I noticed, however, that feature names are removed when passing it somewhere, causing a warning that the model was trained with feature names, and additionally, I see that the generation takes quite a long time (24s on average per record), considering the simple model. Do you have any suggestions on removing the warning and speeding up the generation? I use the Adult dataset if you would like to reproduce that case.
